CONTRIBUTING.md

```diff
@@ -1,15 +1,158 @@
 # Contributing to Zanata
 
+We would love for you to contribute to our source code and to make Zanata
+even better than it is today! Here are the guidelines we would like you to
+follow:
+
+## <a name="Feedback_or_Issues">Feedback or Issues</a>
+If you find a bug, want a feature, or generally want to express your feeling
+toward Zanata, please go to our [issue system](https://zanata.atlassian.net/)
+
+## <a name="Pull_Requests">Pull Requests</a>
 Pull requests welcome!
 
-General information about our Maven setup is in the wiki:
-https://github.com/zanata/zanata-server/wiki/Developer-Guide
+### <a name="Setup">Setup</a>
+The
+[Developer Setup Guide](https://github.com/zanata/zanata-server/wiki/Developer-Guide)
+shows the dependencies and how to setup Java, Maven, MySQL, JBoss, and IDE.
 
-Our CI server merges changes from `integration/master` to `master` after testing, so: **please branch from `master`, but please target `integration/master` in pull requests.**
+### Branches
+The branches that should be targeted in pull requests
+ - **master**: New features, bug fix, enhancement should
+ target this branch. It is essentially master branch before test. If you are
+ unsure which branch to make pull request, use this branch.
+ - **release**: Only urgent bug fix and documentation should target this
+ branch.
 
+### Test against Wildfly
 To run all tests against WildFly (takes about 1 hour):
 
-    mvn clean verify -Dappserver=wildfly8 -DstaticAnalysis -Dchromefirefox
+mvn clean verify -Dappserver=wildfly8 -DstaticAnalysis -Dchromefirefox
+
+### <a name="Commit_Message_Format"> Commit Message Format</a>
+We are inspired the [angular commit message format]
+(https://github.com/angular/angular.js/blob/master/CONTRIBUTING.md#commit),
+that is:
```

Comment:
Is this all just copied verbatim from the angular commit message guidelines? If it is not changed, I think we should just link to it and show an example or two.